standalone/src/models/product-selection/builders.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { TProductSelectionGraphql, TProductSelectionRest } from './types';
+import {
+  ProductSelection,
+  ProductSelectionGraphql,
+  ProductSelectionRest,
+} from './index';
+
+const validateRestModel = (model: TProductSelectionRest): void => {
+  expect(model).toMatchObject({
+    version: expect.any(Number),
+    id: expect.any(String),
+    createdAt: expect.any(String),
+    lastModifiedAt: expect.any(String),
+    lastModifiedBy: expect.objectContaining({
+      externalUserId: expect.any(String),
+      anonymousId: expect.any(String),
+      clientId: expect.any(String),
+      customer: expect.objectContaining({
+        id: expect.any(String),
+        typeId: 'customer',
+        obj: expect.objectContaining({
+          id: expect.any(String),
+        }),
+      }),
+    }),
+    createdBy: expect.objectContaining({
```

Comment:
again just a question what, why dont we check for `attributedTo`?

Reply:
Usually in the REST models we don't test for all the properties of a nested model.
We just test one or two to make sure the shape looks like what we are expecting.

You need to understand the purpose here is just to make sure the property has the expected nested model. With GraphQL it's easy because we can rely on the `__typename` property. For REST, we use some kind of [duck typing](https://en.wikipedia.org/wiki/Duck_typing) for that. If a couple of properties match, that's enough. We think asserting for all of them is too much of a boilerplate and we rely on the nested models own tests to validate its structure.

Reply:
thanks for the explanation @CarlosCortizasCT in both cases 🙇